pkg/qemu/qemu.go

```diff
@@ -285,7 +356,8 @@ func Cmdline(cfg Config) (string, []string, error) {
 		// QEMU <  7.0 requires highmem=off to be set, otherwise fails with "VCPU supports less PA bits (36) than requested by the memory map (40)"
 		// https://github.com/lima-vm/lima/issues/680
 		// https://github.com/lima-vm/lima/pull/24
-		if !strings.Contains(string(features.MachineHelp), "virt-7.0") {
+		// But when the memory size is less than 4 GiB, we can always set highmem=off.
+		if !features.VersionGEQ7 || memBytes < (1<<32) {
```

Comment:
And I wonder if this one is correct too; I couldn't find a definitive source, but I've seen comments that `highmem=off` limits memory on ARM to 3GiB.

Reply:
adjusted to 3 GiB